llvm/include/llvm/CAS/OnDiskGraphDB.h

```diff
@@ -278,7 +278,12 @@ class OnDiskGraphDB {
   /// Returns \p nullopt if the object is not stored in this CAS.
   std::optional<ObjectID> getExistingReference(ArrayRef<uint8_t> Digest);
 
-  /// \returns true if the object associated with \p Ref is stored in the CAS.
+  /// Check whether the object associated with \p Ref is stored in the CAS.
+  /// Note that this function will fault-in according to the policy.
+  Expected<bool> isMaterialized(ObjectID Ref);
+
+  /// Check whether the object associated with \p Ref is stored in the CAS.
+  /// Note that this function does not fault-in.
   bool containsObject(ObjectID Ref) const {
     return containsObject(Ref, /*CheckUpstream=*/true);
```

Comment:
This is a bit too clever and magic 😅 I'd suggest to use a switch on the return value, to make it more readable and for future maintenance in case we add some new enumerator for any reason.